File: js/413.arithmetic-slices.js
/*
 * @lc app=leetcode id=413 lang=javascript
 *
 * [413] Arithmetic Slices
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @return {number}
 */
 var numberOfArithmeticSlices = function(A) {
    let i = 0, j = 2;
    let count = 0;
    let gap = A[1] - A[0];

    while(j < A.length) {
        const currentGap = A[j] - A[j-1];

        if(currentGap === gap) {
            count += (j - i - 1);
        } else {
            i = j - 1;
            gap = currentGap;
        }
        j++
    }
    return count;
};
// @lc code=end

File: kotlin/413.arithmetic-slices.kt
/*
 * @lc app=leetcode id=413 lang=kotlin
 *
 * [413] Arithmetic Slices
 */

// @lc code=start
class Solution {
    fun numberOfArithmeticSlices(nums: IntArray): Int {
        if(nums.size < 3) return 0

        var i = 0
        var j = 2
        var count = 0;

        var gap = nums[1] - nums[0];

        while(j < nums.size) {
            val currentGap = nums[j] - nums[j-1];

            if(currentGap.equals(gap)) {
                count += (j - i - 1);
            } else {
                i = j - 1;
                gap = currentGap;
            }
            j++
        }
        return count;
    }
}
// @lc code=end

